tests/v2/utils/library/test_utils.go

```
@@ -0,0 +1,322 @@
+				}
+			}
+
+			validator.WriteAndValidate(&testDevfile)
+		}
+	}
+}
+
+// verify verifies the library contents of the specified devfile with the expected content
+func verify(devfile *commonUtils.TestDevfile) error {
+
+	commonUtils.LogInfoMessage(fmt.Sprintf("Verify %s : ", devfile.FileName))
+
+	var errorString []string
+
+	libraryData := devfile.Follower.(DevfileFollower).LibraryData
+	commonUtils.LogInfoMessage(fmt.Sprintf("Get commands %s : ", devfile.FileName))
+	commands, _ := libraryData.GetCommands(common.DevfileOptions{})
+	if commands != nil && len(commands) > 0 {
+		err := VerifyCommands(devfile, commands)
+		if err != nil {
+			errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Verfify Commands %s : %v", devfile.FileName, err)))
+		}
+	} else {
+		commonUtils.LogInfoMessage(fmt.Sprintf("No command found in %s : ", devfile.FileName))
+	}
+
+	commonUtils.LogInfoMessage(fmt.Sprintf("Get components %s : ", devfile.FileName))
+	components, _ := libraryData.GetComponents(common.DevfileOptions{})
+	if components != nil && len(components) > 0 {
+		err := VerifyComponents(devfile, components)
+		if err != nil {
+			errorString = append(errorString, commonUtils.LogErrorMessage(fmt.Sprintf("Verfify Commands %s : %v", devfile.FileName, err)))
+		}
+	} else {
+		commonUtils.LogInfoMessage(fmt.Sprintf("No components found in %s : ", devfile.FileName))
+	}
+
+	var returnError error
+	if len(errorString) > 0 {
+		returnError = errors.New(fmt.Sprint(errorString))
+	}
+	return returnError
+
+}
+
+// editCommands modifies random attributes for each of the commands in the devfile.
+func editCommands(devfile *commonUtils.TestDevfile) error {
+
+	commonUtils.LogInfoMessage(fmt.Sprintf("Edit %s : ", devfile.FileName))
+
+	var err error
+	commonUtils.LogInfoMessage(fmt.Sprintf(" -> Get commands %s : ", devfile.FileName))
+	commands, _ := devfile.Follower.(DevfileFollower).LibraryData.GetCommands(common.DevfileOptions{})
```

should not ignore err, same for `editComponents`

---

Fixed